esp32_s2_kaluga_kit/esp32_s2_kaluga_kit.c

```diff
@@ -1,256 +1,302 @@
+        .shield_driver = TOUCH_PAD_SHIELD_DRV_L0,   /*!< 40pf */
+    };
+    ESP_ERROR_CHECK(touch_pad_waterproof_set_config(&waterproof));
+    ESP_ERROR_CHECK(touch_pad_waterproof_enable());
+
+    /*!< Filter setting */
+    touch_filter_config_t filter_info = {
+        .mode = TOUCH_PAD_FILTER_IIR_8,           /*!< Test jitter and filter 1/4. */
+        .debounce_cnt = 1,      /*!< 1 time count. */
+        .noise_thr = 0,         /*!< 50% */
+        .jitter_step = 4,       /*!< use for jitter mode. */
+    };
+    ESP_ERROR_CHECK(touch_pad_filter_set_config(&filter_info));
+    ESP_ERROR_CHECK(touch_pad_filter_enable());
+    /*!< Register touch interrupt ISR, enable intr type. */
+    ESP_ERROR_CHECK(touch_pad_isr_register(fn, NULL, TOUCH_PAD_INTR_MASK_ALL));
+    ESP_ERROR_CHECK(touch_pad_intr_enable(TOUCH_PAD_INTR_MASK_ACTIVE | TOUCH_PAD_INTR_MASK_INACTIVE));
+
+    /*!< Enable touch sensor clock. Work mode is "timer trigger". */
+    ESP_ERROR_CHECK(touch_pad_set_fsm_mode(TOUCH_FSM_MODE_TIMER));
+    ESP_ERROR_CHECK(touch_pad_fsm_start());
+
+    /*!< Wait touch sensor init done and calibrate */
+    vTaskDelay(100 / portTICK_PERIOD_MS);
+    for (int i = 0; i < TOUCH_BUTTON_NUM; i++) {
+        bsp_touchpad_calibrate(bsp_touch_button[i], 0.1f);
+    }
+}
+
+void bsp_touchpad_calibrate(bsp_touchpad_button_t tch_pad, float tch_threshold)
+{
+    /*!< read baseline value */
+    uint32_t touch_value = 0;
+    ESP_ERROR_CHECK(touch_pad_read_benchmark(tch_pad, &touch_value));
+    /*!< set interrupt threshold. */
+    ESP_ERROR_CHECK(touch_pad_set_thresh(tch_pad, (uint32_t)((float)touch_value * tch_threshold)));
+}
+
+// Bit number used to represent command and parameter
+#define LCD_CMD_BITS           (8)
+#define LCD_PARAM_BITS         (8)
+static lv_disp_draw_buf_t disp_buf; // contains internal graphic buffer(s) called draw buffer(s)
+static lv_disp_drv_t disp_drv;      // contains callback functions
+
+static bool lvgl_port_flush_ready(esp_lcd_panel_io_handle_t panel_io, esp_lcd_panel_io_event_data_t *edata, void *user_ctx)
+{
+    lv_disp_drv_t *disp_driver = (lv_disp_drv_t *)user_ctx;
+    lv_disp_flush_ready(disp_driver);
+    return false;
+}
+
+static void lvgl_port_flush_callback(lv_disp_drv_t *drv, const lv_area_t *area, lv_color_t *color_map)
+{
+    esp_lcd_panel_handle_t panel_handle = (esp_lcd_panel_handle_t) drv->user_data;
+    const int offsetx1 = area->x1;
+    const int offsetx2 = area->x2;
+    const int offsety1 = area->y1;
+    const int offsety2 = area->y2;
+    // copy a buffer's content to a specific area of the display
+    esp_lcd_panel_draw_bitmap(panel_handle, offsetx1, offsety1, offsetx2 + 1, offsety2 + 1, color_map);
+}
+
+#if defined(CONFIG_BSP_DISPLAY_CREATE_LVGL_TASK)
+static void lvgl_port_task(void *arg)
+{
+    while (1) {
+        vTaskDelay(pdMS_TO_TICKS(10));
+        // Be carefull here, this call is not thread safe
+        lv_timer_handler();
```

Comment:
The idea was that in case user wants to be thread-safe, he disables `CONFIG_BSP_DISPLAY_CREATE_LVGL_TASK` in menuconfig and then the user is responsible for calling `lv_timer_handler()` in his application.

This is going to be changed. We'll provide only the clean thread-safe option.